src/status.h

```diff
@@ -43,10 +46,15 @@ struct Status {
 /// human-readable strings to stdout
 struct StatusPrinter : Status {
   explicit StatusPrinter(const BuildConfig& config);
-  virtual void PlanHasTotalEdges(int total);
+
+  /// Callbacks for the Plan to notify us about adding/removing Edge's.
+  virtual void EdgeAddedToPlan(const Edge* edge);
```

Comment:
These functions override the function in the base class.

Use the "override" keyword.

If they're the most-derived function in the codebase for this class hierarchy, also use the "final" keyword to help the compiler perform function-devirtualization (important for performance)

Reply:
If i change these, the rest of the `virtual`-that-should-be-`override` get diagnosed.
I think that cleanup should be done separately from this change.

Reply:
How are they not already getting diagnosed? The compiler being overly helpful?

I think you should just do your new code correctly for C++11, and not worry about pre-existing code having new warnings. The codebase already has warnings anyway.

But like I said, i have no control over committing / merging, so do whatever gets your change merged the quickest.